compose/src/main/java/com/bottlerocketstudios/compose/devoptions/DevOptions.kt

```diff
@@ -0,0 +1,317 @@
+            modifier = Modifier
+                .padding(start = 8.dp, end = 8.dp, top = 12.dp, bottom = 12.dp)
+        ) {
+            content()
+        }
+    }
+}
+
+// TODO this needs to be update when the next stable release comes out. New Menus
+@Composable
+@Suppress("LongMethod")
+private fun DropdownEnvMenu(state: DevOptionsState) {
+    var expanded by remember { mutableStateOf(false) }
+    val items = state.environmentNames.value
+    Box(
+        modifier = Modifier
+            .fillMaxWidth()
+            .wrapContentSize(Alignment.TopStart)
+    ) {
+        items[state.environmentSpinnerPosition.value].let {
+            Row(
+                Modifier
+                    .fillMaxWidth()
+                    .clickable(onClick = { expanded = true })
+                    .height(IntrinsicSize.Min)
+                    .background(
+                        Color.LightGray
+                    )
+            ) {
+                Column(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .wrapContentHeight(Alignment.CenterVertically)
+                        .weight(1.0F)
+                        .padding(12.dp)
+                ) {
+                    Text("Environment Switcher", style = TextStyle(color = if (expanded) MaterialTheme.colors.primary else Color.DarkGray, fontSize = 12.sp))
+                    Text(it, style = TextStyle(color = Color.Black, fontSize = 14.sp))
+                }
+
+                val displayIcon: Painter = painterResource(
+                    id = R.drawable.ic_arrow_drop_down_24
+                )
+                Icon(
+                    painter = displayIcon,
+                    contentDescription = null,
+                    modifier = Modifier
+                        .wrapContentWidth(Alignment.CenterHorizontally)
+                        .fillMaxHeight()
+                        .padding(end = 12.dp),
+                    tint = if (expanded) MaterialTheme.colors.primary else Color.DarkGray
+                )
+            }
+        }
+        DropdownMenu(
+            expanded = expanded,
+            onDismissRequest = { expanded = false },
+            modifier = Modifier
+                .fillMaxWidth()
+                .background(
+                    Color.White
+                )
+        ) {
+            items.forEachIndexed { index, s ->
+                DropdownMenuItem(onClick = {
+                    expanded = false
+                    state.onEnvironmentChanged(index)
+                }) {
+                    Text(text = s, style = TextStyle(color = Color.Black))
+                }
+            }
+        }
+    }
+}
+
+@Preview(showBackground = true)
+@Composable
+private fun PreviewOuterScreenContent() {
+    val spinnerPosition = remember { mutableStateOf(0) }
+    DevOptionsScreen(
+        state = DevOptionsState(
+            environmentNames = remember { mutableStateOf(listOf("POC", "TST", "PROD")) },
```

Comment:
Huge improvement over creating/injecting fake data/no-op implementations for the ViewModel 💯